src/compiler/checker.ts

```diff
@@ -19533,7 +19494,7 @@ namespace ts {
                 priority = savePriority;
             }
 
-            function invokeOnce(source: Type, target: Type, action: (source: Type, target: Type) => void) {
+            function invokeWithDepthLimit(source: Type, target: Type, action: (source: Type, target: Type) => void) {

```

Comment:
This is pretty much the inference equivalent of `recursiveTypeRelatedTo` at this point.